packages/vendure-plugin-shipping-extensions/src/services/zone-aware-shipping-tax-calculation.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import {
+  ID,
+  Logger,
+  Order,
+  RequestContext,
+  TaxRate,
+  TransactionalConnection,
+} from '@vendure/core';
+import { loggerCtx } from '../constants';
+
+@Injectable()
+export class ZoneAwareShippingTaxCalculationService {
+  constructor(private readonly connection: TransactionalConnection) {}
+
+  async getTaxRateWithCategory(
```

Comment:
Shouldn't this `getTaxRateForCategory`? Maybe also add a comment with something like "This resolves the given tax category to a tax rate based on the orders billing Address. If no billing address, it used the shipping address"